docs/Coding_Style_Guide.md

```diff
@@ -1,17 +1,357 @@
+{
+...
+}
+```
+
+Calling other constructors inside a constructor should be avoided.
+This is in preparation for an environment where we cannot use c++11 features (such as embedded devices).
+```cpp
+Dog::Dog()
+    , m_isAnimal(true)
+{
+}
+
+Dog::Dog(String name, Breed breed)
+    : Dog()  //<--- Not allowed
+    , m_name(name)
+    , m_breed(breed)
+{
+...
+}
+
+```
+
+Write it one by one.
+```cpp
+Dog::Dog()
+    , m_isAnimal(true)
+    , m_name(String::emptyString)
+    , m_breed(Breed::emptyBreed)
+{
+}
+
+Dog::Dog(String name, Breed breed)
+    : m_isAnimal(true)
+    , m_name(name)
+    , m_breed(breed)
+{
+...
+}
+
+```
+
+
+### Class Modifiers
+Do not indent class modifiers (i.e., `public`, `protected`, and `private`) in
+a class declaration.
+
+```cpp
+class Dog {
+public:
+    ...
+private:
+    ...
+}
+```
+
+## Functions
+### Function Names
+Use camelcases when naming a function.
+
+### Function Calls
+Write a function call all in the same line if it fits. If not, split the
+function call into multiple lines by either adding a newline after the
+assignment operator or between function parameters. When splitting between
+parameters, align them with the first parameter. In addition, do not add spaces
+before and after the first and last function parameter, respectively.
+
+```cpp
+int val = foo(arg1, arg2, arg3);
+
+int val =
+    foo(arg1, arg2, arg3);
+
+int val = foo(arg1, arg2
+              arg3);
+```
+
+### Function Declaration and Definition
+Use named parameters in function declarations.
 
+```cpp
+returnType functionName(int, int); // Not allowed
+returnType functionName(int arg1, int arg2);  // Use named parameters
+```
+
+Return type should be on the same line as the function declaration
+(and definition) if it fits. If not, split function parameters into multiple
+lines, and align them with the first parameter.
+
+```cpp
+returnType functionName(int arg1,
+                        int arg2);
+```
+
+If the first parameter does not fit in a line, write parameters in the next
+line with 4 space indent.
+
+```cpp
+returnType functionName(
+    int arg1, int arg2);
+```
+
+The opening curly brace should be on its own in the next line.
+A Closing curly brace should be on the next line as the opening brace.
+
+```cpp
+returnType functionName()
+{
+    ...
+}
+```
+
+### Short and Empty Functions
+Do not write a function all in one line even for a very short function that
+can be fit in one line. One exception is an empty, inline function that is
+declared in a header file.
+
+```cpp
+void f() { } // Allowed only in a header file
+
+int g()
+{
+    return 0;
+}
+```
+
+## C++ Features
+### Run-Time Type Information (RTTI)
+Do not use Run Time Type Information.
+
+### C++11
+Use of C++11 features are encouraged. In addition, use of C++11 compatible
+style formatting is encouraged, e.g., use `A<B<int>>` instead of `A<B<int> >`
+
+### Use of `try-catch` statements
+Do not use try-catch statements except throwing a DOMException.
```

Comment:
This line is not suitable for escargot but starfish

Reply:
Fixed it from `DOMException` to `Exception`. Thanks.